src/org/jitsi/impl/neomedia/ByteArrayBufferImpl.java

```diff
@@ -0,0 +1,149 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.impl.neomedia;
+
+import org.jitsi.service.neomedia.*;
+
+import java.util.*;
+
+/**
+ * Implements {@link ByteArrayBuffer}.
+ * @author Boris Grozev
+ */
+public class ByteArrayBufferImpl
+    implements ByteArrayBuffer
+{
+    /**
+     * The byte array represented by this {@link ByteArrayBufferImpl}.
+     */
+    private byte[] buffer;
+
+    /**
+     * The offset in the byte buffer where the actual data starts.
+     */
+    private int offset;
+
+    /**
+     * The length of the data in the buffer.
+     */
+    private int length;
+
+    /**
+     * Initializes a new {@link ByteArrayBufferImpl} instance.
+     * @param buffer
+     * @param offset
+     * @param length
+     */
+    public ByteArrayBufferImpl(byte[] buffer, int offset, int length)
+    {
+        this.buffer = Objects.requireNonNull(buffer, "buffer");
+        if (offset + length > buffer.length)
+        {
+            throw new IllegalArgumentException("length");
+        }
+        this.offset = offset;
+        this.length = length;
+    }
+
+    /**
+     * Initializes a new {@link ByteArrayBufferImpl} based on a newly allocated
+     * byte array with the given size.
+     * @param size the size of the underlying byte array.
+     */
+    public ByteArrayBufferImpl(int size)
+    {
+        buffer = new byte[size];
+        offset = 0;
+        length = size;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public byte[] getBuffer()
+    {
+        return buffer;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getOffset()
+    {
+        return offset;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getLength()
+    {
+        return length;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setLength(int length)
+    {
+        if (offset + length > buffer.length)
+        {
+            throw new IllegalArgumentException("length");
+        }
+        this.length = length;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setOffset(int offset)
+    {
+        if (offset + length > buffer.length)
+        {
+            throw new IllegalArgumentException("length");
+        }
+        this.offset = offset;
+    }
+
+    /**
+     * Sets the offset and the length of this {@link ByteArrayBuffer}
+     * @param offset the offset to set.
+     * @param length the length to set.
+     */
+    public void setOffsetLength(int offset, int length)
+    {
+        if (offset + length > buffer.length)
+        {
+            throw new IllegalArgumentException("length");
+        }
+        this.offset = offset;
+        this.length = length;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isInvalid()
+    {
+        return false;
```

Comment:
These conditions are enforced by the public API, so it would be equivalent to "return false", unless there's a bug somewhere. Will add it if you still think it should be there.